### ISSUE TITLE
stopTimeout option not showing in inspect output

### ISSUE THREAD

**Issue by keirwhitlock** [aws/amazon-ecs-agent#2879]
### Summary
Setting the stopTimeout in the TaskDefinition, isn't getting passed in as an option to the starting of the container.

### Description
I am setting a stopTimeout in my TaskDefinition of 1800s; which shows in the JSON. However when the ECS service requests a task to run, the agent doesn't appear to be passing in this option to the starting of the container.

Running a `docker inspect <container>` doesn't show the option in the output.

Starting a container manually using `docker run` with the `--stop-timeout='1800'` option; does show in the inspect output for the container.

### Expected Behavior
`"StopTimeout": 1800` to show in the `docker inspect` output; if set in the TaskDefinition.

### Observed Behavior
No value shown in the inspect output.

### Environment Details
* docker info
```
Client:
 Debug Mode: false

Server:
 Containers: 5
  Running: 5
  Paused: 0
  Stopped: 0
 Images: 6
 Server Version: 19.03.13-ce
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: c623d1b36f09f8ef6536a057bd658b3aa8632828
 runc version: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
 init version: de40ad0 (expected: fec3683)
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.14.231-173.361.amzn2.x86_64
 Operating System: Amazon Linux 2
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.909GiB
 Name: ip-10-10-1-101.eu-west-2.compute.internal
 ID: B6FL:VFMV:GYYG:WABA:LXV7:U4ZN:FYT3:7MYX:RCHZ:GORP:UAGR:5FUK
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```
* curl http://localhost:51678/v1/metadata
`"Version":"Amazon ECS Agent - v1.52.2 (fddb17a1)"`

**Comment by shubham2892** [comment]
@keirwhitlock can you share the task def you are using? Are you setting the value to `1800` or `1800s` ?

**Comment by keirwhitlock** [comment]
> @keirwhitlock can you share the task def you are using? Are you setting the value to `1800` or `1800s` ?

Hi @shubham2892 - Sorry for the ambiguity. It's 1800. The AWS console wont actually allow a non integer value.

![image](https://user-images.githubusercontent.com/75618555/119490227-8323dc00-bd54-11eb-8e9a-20bf181f1491.png)

**Comment by Phelan164** [comment]
@keirwhitlock  stopTimeout working or not?
I had issue stopTimeout not working (grace period only ~10 seconds)

**Comment by keirwhitlock** [comment]
> @keirwhitlock stopTimeout working or not?
> I had issue stopTimeout not working (grace period only ~10 seconds)

It appears to be. It doesn't show in the inspect output, but does appear to be respecting it. The issue I was having drilled down to the CDK I was using, is setting up the ASG lifecycle hook for me; and that code had a bug in it. In the end I wrote my own lambda function and lifecycle hook into my CDK, rather than using the one auto generated.

In their defence, whoever wrote the code that is included in the CDK provided lifecycle; the option I used was deprecated, so I can't complain.
